aspnet/testing/integration-testing/sample/src/PrimeWeb/Middleware/PrimeCheckerMiddleware.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using PrimeWeb.Services;
+using System;
+using System.Threading.Tasks;
+
+namespace PrimeWeb.Middleware
+{
+    public class PrimeCheckerMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly PrimeCheckerOptions _options;
+
+        public PrimeCheckerMiddleware(RequestDelegate next,
+            PrimeCheckerOptions options)
+        {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
+            }
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            _next = next;
+            _options = options;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            HttpRequest request = context.Request;
+            if (!request.Path.HasValue ||
+                request.Path != _options.Path)
+            {
+                await _next.Invoke(context);
+            }
+            else
+            {
+                int numberToCheck;
+                try
+                {
+                    numberToCheck = int.Parse(request.QueryString.Value.Replace("?", ""));
+                    var primeService = new PrimeService();
```

Comment:
If this is really a service it should be injected via DI, not created per request.